features/192-asset-gear/index.md

```diff
@@ -0,0 +1,640 @@
+# Asset Gear
+
+## Goals
+
+- Store user-generated asset to fulfill application needs.
+  - Photos and short videos taken by the application.
+  - Images and short videos chosen from the library.
+  - Artwork by the user.
+  - Attachments in a chat application.
+  - Documents of reasonable size.
+- Process image asset on the fly via query string.
+- Access control via presigned URL
+
+## Non-goals
+
+- Video live streaming such as Facebook Live.
+- File sharing application like Imgur.
+- Static website hosting. Another gear is responsible for this.
+
+## Layout
+
+The asset name is in the following format.
+
+```
+/<app-id>/<random-string>[.<ext>]
+```
+
+- `app-id`: The app ID.
+- `random-string`: A randomly generated string.
+- `ext`: The inferred file extension (if any) from the given content type.
+
+## HTTP API
+
+### POST /_asset/presign_upload
+
+#### Description
+
+This endpoint returns a presigned upload request which can be used to upload an asset.
+
+This endpoint requires API Key and authenticated user.
+
+#### Request JSON Schema
+
```

Comment:
Curious, is it a standard format to describe JSON?

Reply:
It is [JSON Schema](https://json-schema.org/). It is still an IETF draft. We are now migrating our hand-written validation code to JSON Schema.